src/compression/compressor.rs

```diff
@@ -0,0 +1,56 @@
+use super::serialize::node_to_bytes_backrefs;
+use clvmr::allocator::{Allocator, NodePtr};
+use clvmr::chia_dialect::ChiaDialect;
+use clvmr::node::Node;
+use clvmr::run_program::run_program;
+use clvmr::serialize::{node_from_bytes, node_to_bytes};
+
+fn wrap_atom_with_decompression_program(
+    allocator: &mut Allocator,
+    node_ptr: NodePtr,
+) -> Result<NodePtr, std::io::Error> {
+    let apply_node = allocator.new_atom(&[2])?;
+    let quote_node = allocator.one();
+    let serialized_backrefs_program = include_bytes!("deserialize_w_backrefs.bin");
+    // "(a (q . deserialize_w_backrefs_program) (q . serialized_with_backrefs))"
+    let program = node_from_bytes(allocator, serialized_backrefs_program)
+        .expect("can't deserialize backref prog");
+
+    let compressed_block = allocator.new_pair(quote_node, node_ptr)?;
+    let program = allocator.new_pair(quote_node, program)?;
+    let list = allocator.null();
+    let list = allocator.new_pair(compressed_block, list)?;
+    let list = allocator.new_pair(program, list)?;
+    let list = allocator.new_pair(apply_node, list)?;
+    Ok(list)
+}
+
+pub fn decompress(allocator: &mut Allocator, blob: &[u8]) -> Result<NodePtr, std::io::Error> {
```

Comment:
if we ever end up needing this in the node or wallet, I would expect we'd want a rust implementation.